### PR TITLE
fix(core,platform): schematics, deps, migrations, assets & styles

### PR DESCRIPTION
**Pull request by platon-rov**
Original PR https://github.com/SAP/fundamental-ngx/pull/8465 **`<= Please check comments there also`**

## Related Issue(s)

Closes https://github.com/SAP/fundamental-ngx/issues/8350
Closes https://github.com/SAP/fundamental-ngx/issues/8457
Refers to https://github.com/SAP/fundamental-ngx/pull/8258

## Description

* Library deps were made peerDeps to fix various issues with multiple deps instances.
* Core schematics updated to setup Theming out of the box.
* Core schematics styles and assets adding fixed to have all the files only on the library (our) side.
* Migrations created to remove old styles and assets.

**TDB**, Not done
* Removing the old deprecated Themes.

##### PR Quality

-   [X] [Breaking Changes Wiki](https://github.com/SAP/fundamental-ngx/wiki/0.36.0-Breaking-changes#schematics-8465)

### CODE REVIEW

File: libs/core/schematics/migrations/migration-0.36/index.ts [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+
+                    tree.overwrite(styleSheet, contents);
+                    context.logger.info(`✅️ Removed font styles from stylesheets.`);
+                }
+            });
+        }
+
+        context.logger.info(
+            `⚠️ Notice: If you have imported font styles from @sap-theming, please remove it since now it controlled by fundamental-ngx`
+        );
```
Comment (g-cheishvili) [inline libs/core/schematics/migrations/migration-0.36/index.ts:117]:
```suggestion
        context.logger.info(
            `⚠️ Notice: If you have imported font styles from @sap-theming, please remove it, since now it is controlled by fundamental-ngx`
        );
```

File: libs/core/schematics/migrations/migration-0.36/index.ts [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+
+        return tree;
+    };
+}
+
+function noticeAddSchematics(): Rule {
+    return (_: Tree, context: SchematicContext) => {
+        context.logger.info(`ℹ️ Now you have run ng-add schematics once again to setup things in the right way.`);
+
```
Comment (g-cheishvili) [inline libs/core/schematics/migrations/migration-0.36/index.ts:126]:
```suggestion
    return (_: Tree, context: SchematicContext) => {
        context.logger.info(`ℹ️ Now you have to run ng-add schematics once again to set up things in the right way.`);

```

File: libs/core/schematics/utils/file-utils.ts [hunk 3]
```diff
@@ -1,10 +1,11 @@
 import { join, Path } from '@angular-devkit/core';
 import { Tree } from '@angular-devkit/schematics';
 
-const STYLESHEET_REGEX = /.*\.(css|scss|less)/;
+const STYLESHEET_REGEX = /.*\.(css|scss)/;
```
Comment (g-cheishvili) [inline libs/core/schematics/utils/file-utils.ts:4]: What do we have against "sass" and "less" extensions?

File: package.json [hunk 4]
```diff
@@ -73,8 +73,8 @@
     "compile-typedoc-platform": "rm -rf apps/docs/src/assets/typedoc/platform && typedoc --out apps/docs/src/assets/typedoc/platform libs/platform/src/lib --tsconfig libs/platform/src/lib/tsconfig.lib.json --hideGenerator --theme apps/docs/src/fd-typedoc",
     "compile-typedoc-json": "typedoc --json apps/docs/src/assets/typedoc/typedoc.json --hideGenerator --theme apps/docs/src/fd-typedoc libs/core/src/lib",
     "build:schematics": "./node_modules/.bin/tsc -p libs/core/tsconfig.schematics.json && yarn run copy:schemas && yarn run copy:collection",
-    "copy:schemas": "cp libs/core/schematics/ng-add/schema.json dist/libs/core/schematics/ng-add/schema.json && yarn run copy:core-migration-shema",
-    "copy:core-migration-shema": "cp libs/core/schematics/migrations/migrate-theme-fonts/schema.json dist/libs/core/schematics/migrations/migrate-theme-fonts/schema.json",
+    "copy:schemas": "cp libs/core/schematics/ng-add/schema.json dist/libs/core/schematics/ng-add/schema.json && yarn run copy:core-migration-schema",
+    "copy:core-migration-schema": "cp libs/core/schematics/migrations.json dist/libs/core/schematics/migrations.json",
```
Comment (g-cheishvili) [inline package.json:77]: Thanks. without bash I will not be able to use these scripts on my machine even more(Windows...)

**Review by g-cheishvili — APPROVED** [review]
Only grammatical issues, nothing more

**Review by g-cheishvili — CHANGES_REQUESTED** [review]
See above

File: package.json (hunk 4 above)
Comment (platon-rov) [reply package.json:77]: actually I didn't change anything, just file names

Comment (platon-rov) [reply package.json:77]: in case we have issues with commands we could fix it in separate pr.

File: libs/core/schematics/utils/file-utils.ts (hunk 3 above)
Comment (platon-rov) [reply libs/core/schematics/utils/file-utils.ts:4]: nothing :D 

fixed

File: libs/platform/src/lib/package.json [hunk 5]
```diff
@@ -19,12 +19,13 @@
     "@angular/animations": "ANGULAR_VER_PLACEHOLDER",
     "@angular/router": "ANGULAR_VER_PLACEHOLDER",
     "@angular/cdk": "ANGULAR_VER_PLACEHOLDER",
-    "rxjs": "RXJS_VER_PLACEHOLDER"
+    "rxjs": "RXJS_VER_PLACEHOLDER",
+    "fundamental-styles": "FDSTYLES_VER_PLACEHOLDER",
```
Comment (InnaAtanasova) [inline libs/platform/src/lib/package.json:23]: do we need the FD prefix?

Comment (g-cheishvili) [reply libs/platform/src/lib/package.json:23]: Do not think we need, otherwise prepare executor should be updated as well to properly handle placeholder @platon-rov

File: libs/core/src/lib/package.json [hunk 6]
```diff
@@ -19,16 +22,21 @@
...
-    "fundamental-styles": "FDSTYLES_VER_PLACEHOLDER",
     "@fundamental-ngx/i18n": "VERSION_PLACEHOLDER",
     "fast-deep-equal": "FAST_DEEP_EQUAL_VER_PLACEHOLDER",
     "lodash-es": "LODASH_ES_VER_PLACEHOLDER",
     "tslib": "^2.0.0",
     "compare-versions": "COMPARE_VERSIONS_VER_PLACEHOLDER"
+  },
+  "exports": {
+    ".": {
+      "sass": "./styles/fundamental-ngx-core.scss"
```
Comment (InnaAtanasova) [inline libs/core/src/lib/package.json:39]: why is this needed?

Comment (platon-rov) [reply libs/core/src/lib/package.json:39]: Currently, we have *raw* global styles `libs/core/schematics/ng-add/styles.ts` (fonts, icons, etc.) which we place on the customer side when installing lib with `ng add`. In case tomorrow we need to change these styles, we have to write migration for it. 

If we only be putting link to styles file from our lib, we can change its contents to whatever we want without need to write migrations etc.

Real world example: 
Tomorrow we will add Skeleton feature and these Skeletons will be rendered over the components when applying CSS classes. To have these CSS classes' values loaded, we either have to render a skeleton component somewhere which import these styles (bad approach) or add these styles to global ones (like fonts) so they will be accessible anywhere once global styles are loaded.

File: libs/platform/src/lib/package.json (hunk 5 above)
Comment (platon-rov) [reply libs/platform/src/lib/package.json:23]: What do you mean? I didn't change anything except moving from deps to peerDeps.

File: libs/core/schematics/migrations.json [hunk 7]
```diff
@@ -0,0 +1,9 @@
+{
+    "schematics": {
+        "migration-0-37": {
+            "version": "0.37",
```
Comment (g-cheishvili) [inline libs/core/schematics/migrations.json:4]: Does not this mean that if user updates from anything below 0.37.0 then this schematic will be called, but we're currently on 0.36.3. Am I getting this right?

Comment (platon-rov) [reply libs/core/schematics/migrations.json:4]: This schematics update is a breaking change by itself, once we merge this pr `ng add` will start working differently and that's why I set version here to `0.37`, merge of this pr will promote version to `0.37`

**Review by InnaAtanasova — APPROVED** [review]

**Review by g-cheishvili — APPROVED** [review]
